backend/Application/Api/GraphQL/Import/AccountLookup.cs

```diff
@@ -17,16 +21,38 @@ public class AccountLookup : IAccountLookup
     private readonly IMemoryCache _cache;
     private readonly DatabaseSettings _dbSettings;
     private readonly IMetrics _metrics;
+    private readonly IConcordiumNodeClient _client;
     private readonly ILogger _logger;
+    private readonly IBlockHashInput _blockHashInput = new LastFinal();
 
-    public AccountLookup(IMemoryCache cache, DatabaseSettings dbSettings, IMetrics metrics)
+    public AccountLookup(
+        IMemoryCache cache,
+        DatabaseSettings dbSettings,
+        IMetrics metrics,
+        IConcordiumNodeClient client)
     {
         _cache = cache;
         _dbSettings = dbSettings;
         _metrics = metrics;
+        _client = client;
         _logger = Log.ForContext(GetType());
     }
 
+    /// <summary>
+    /// The method first attempts to get account index using base address from cache,
```

Comment:
Very clear description . However, I would suggest starting with a one-liner that describes what the method does, before the next lines go into detail with _how_ it does it. E.g., "Gets account id's from [accountBaseAddresses]" (disclaimer: I don't know the correct markdown for C#. 

PS: A single sentence summary is recommended in Dart documentation, but I guess the principle is rather universal. https://dart.dev/effective-dart/documentation